src/miral/shared_data.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Alan Griffiths <alan@octopull.co.uk>
+ */
+
+#ifndef MIRAL_SHARED_DATA_H
+#define MIRAL_SHARED_DATA_H
+
+#include <string>
+
+
+namespace miral
+{
+// Root filename of the current server.
```

Comment:
Now I see, it's the binary name of the server you're using here.

Using miral-kiosk:
openat(AT_FDCWD, "/home/gerry/.config/miral-kiosk.bin.display", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/xdg/xdg-ubuntu/miral-kiosk.bin.display", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/xdg/miral-kiosk.bin.display", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/gerry/.config/miral-kiosk.bin.config", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/xdg/xdg-ubuntu/miral-kiosk.bin.config", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/xdg/miral-kiosk.bin.config", O_RDONLY) = -1 ENOENT (No such file or directory)

This isn't to my taste - basing config file names on the binary name is unintuitive to me - but it does work and I cannot come up with a good reason why not.

Reply:
> This isn't to my taste - basing config file names on the binary name is unintuitive to me - but it does work and I cannot come up with a good reason why not.

I'm open to suggestions.

For the .config file, this is a fallback when the shell doesn't provide a name. It just happens that none of our shells provide a name.

For the .display file this was "the simplest thing that might work". (But it would be inconsistent for shells that do supply a .config name.)